### PR TITLE
Enhance email logic and update tests with edge cases

### LINKED ISSUE #117: Update Email to check proper email syntaxing during Add/Edit
Currently, only the `@` is accounted for, but not the `.`

This should probably be handled better as it is also easy for our users to miss this out, leading to potentially unexpected behaviour

Image

### PR DESCRIPTION
**Pull request by bipplane**
Previously, emails such as `user@domain` were considered valid. Now, only emails following the syntax `user@domain.com` are considered valid, in order to prevent bugs or typos within the email field.

Fixes #117

### CODE REVIEW

File: src/test/java/seedu/address/model/person/EmailTest.java [hunk 1]
```diff
@@ -51,15 +51,23 @@ public void isValidEmail() {
...
+        assertFalse(Email.isValidEmail("user@xn--80aswg.xn--p1ai")); // punycode domain name
+        assertFalse(Email.isValidEmail("peterjackpeterjackpeterjackpeterjackpeterjackpeterjackpeterjackpeterjack@"
+                + "gmail.comcomcomcomcomcomcomcomcomcomcomcom")); // exceeds max length
 
         // valid email
         assertTrue(Email.isValidEmail("PeterJack_1190@example.com")); // underscore in local part
         assertTrue(Email.isValidEmail("PeterJack.1190@example.com")); // period in local part
         assertTrue(Email.isValidEmail("PeterJack+1190@example.com")); // '+' symbol in local part
         assertTrue(Email.isValidEmail("PeterJack-1190@example.com")); // hyphen in local part
-        assertTrue(Email.isValidEmail("a@bc")); // minimal
```
Comment (izniy) [inline src/test/java/seedu/address/model/person/EmailTest.java:60]: Good use of equivalence partitioning to cover test cases that might be treated differently

File: src/test/java/seedu/address/model/person/EmailTest.java [hunk 2]
```diff
@@ -51,15 +51,23 @@ public void isValidEmail() {
...
         assertFalse(Email.isValidEmail("peterjack@example.com-")); // domain name ends with a hyphen
         assertFalse(Email.isValidEmail("peterjack@example.c")); // top level domain has less than two chars
+        assertFalse(Email.isValidEmail("\"quoted string\"@example.com")); // quoted string in local part
+        assertFalse(Email.isValidEmail(".user@example.com")); // local part starts with a period
+        assertFalse(Email.isValidEmail("user.@example.com")); // local part ends with a period
+        assertFalse(Email.isValidEmail("user+-name@example.com")); // invalid special characters in local part
+        assertFalse(Email.isValidEmail("user@example..com")); // domain name has two consecutive periods
+        assertFalse(Email.isValidEmail("user@xn--80aswg.xn--p1ai")); // punycode domain name
+        assertFalse(Email.isValidEmail("peterjackpeterjackpeterjackpeterjackpeterjackpeterjackpeterjackpeterjack@"
+                + "gmail.comcomcomcomcomcomcomcomcomcomcomcom")); // exceeds max length
```
Comment (izniy) [inline src/test/java/seedu/address/model/person/EmailTest.java:61]: Should we also include a test case where the exceeding of the max length is due to an excessively long portion before the "." instead of after the "."?

**Review by izniy — APPROVED** [review]
Overall LGTM! The comments make the purpose of each test case clearer

File: src/main/java/seedu/address/model/person/Email.java [hunk 3]
```diff
@@ -48,6 +51,13 @@ public Email(String email) {
      * Returns if a given string is a valid email.
      */
     public static boolean isValidEmail(String test) {
+        requireNonNull(test);
+        test = test.trim();
+
+        if (!test.contains("@") || !test.contains(".") || test.length() >= WORD_LIMIT) {
+            return false;
+        }
+
```
Comment (flljy940) [inline src/main/java/seedu/address/model/person/Email.java:60]:
```suggestion
        requireNonNull(test);
        test = test.trim();

        if (!test.contains("@") || !test.contains(".") || test.length() >= WORD_LIMIT) {
            return false;
        }
        
        // Additionally can add this edge case
        if (test.startsWith("@") || test.endsWith("@")) {
        return false;
        }

```

**Review by flljy940 — APPROVED** [review]
looks good!

File: src/test/java/seedu/address/model/person/EmailTest.java (hunk 2 above)
Comment (bipplane) [reply src/test/java/seedu/address/model/person/EmailTest.java:61]: this is probably unnecessary, as the code checks the length of the entire string regardless of where the text is. regardless, good suggestion!

File: src/main/java/seedu/address/model/person/Email.java (hunk 3 above)
Comment (bipplane) [reply src/main/java/seedu/address/model/person/Email.java:60]: great point! will add this into the code.
